### PR TITLE
Release 2 & 3: Expose approver_emails and category_ids in projects post and bulk post request payload

### PR DESCRIPTION
**Pull request by modernwarfareuplink**
## Description

Expose `category_ids` and `approver_emails` key in platform POST and bulk POST projects API.

Clickup link: https://app.clickup.com/t/85zrhfv01

### CODE REVIEW

File: reference/admin.yaml [hunk 1]
```diff
@@ -293,7 +293,52 @@ components:
...
+
+            - The default value for approver_user_ids is `[]` and it is
+            applicable only when creating the project.
+        approver_emails:
+          type: array
+          maxItems: 2
+          nullable: false
+          items:
+            $ref: '#/components/schemas/email'
+          example:
```
Comment (viiicky) [inline reference/admin.yaml:312]: why no example for `approver_user_ids`?

**Review by viiicky — APPROVED** [review]

File: src/admin/paths/admin@projects@bulk.yaml [hunk 2]
```diff
@@ -38,19 +38,19 @@ post:
                     type: string
                     nullable: true
                   data:
-                    type: string
+                    type: object
```
Comment (modernwarfareuplink) [inline src/admin/paths/admin@projects@bulk.yaml:41]: This is the actual change. The `type` has to be object for the bulk errors and this also works fine for other errors

File: src/components/schemas/project.yaml [hunk 3]
```diff
@@ -99,7 +99,43 @@ project_in:
...
+        - $ref: './fields.yaml#/approver_user_ids'
+      description: |
+        - List of IDs of users who are approvers of the project.
+        - Either `approver_user_ids` or `approver_emails` can be used but not both.
+        - The default value for approver_user_ids is `[]` and it is applicable only when creating the project.
+    approver_emails:
+      type: array
+      maxItems: 2
+      nullable: False
+      items:
```
Comment (modernwarfareuplink) [inline src/components/schemas/project.yaml:112]: I have not re-used the `email` definition from the fields.yaml file because there is some bug in openapi library where the `nullable` property cannot be overriden.

TLDR version: `allOf` and `nullable` doesn't work together

**Comment by modernwarfareuplink** [comment]
Before this PR:
Post Projects
Before the PR

Bulk Post Projects
Before the PR bulk


After this PR:
Post Projects
After the PR

Bulk Post Projects
After the PR bulk

**Review by viiicky — APPROVED** [review]

**Comment by viiicky** [comment]
noice; the screenshot attached was helpful
